systemvm/patches/debian/config/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -452,9 +452,9 @@ def fw_vpcrouter(self):
             self.fw.append(
                 ["filter", "", "-A INPUT -i %s -p udp -m udp --dport 67 -j ACCEPT" % self.dev])
             self.fw.append(
-                ["filter", "", "-A INPUT -i %s -p udp -m udp --dport 53 -j ACCEPT" % self.dev])
+                ["filter", "", "-A INPUT -i %s -p udp -m udp --dport 53 -s %s -j ACCEPT" % (self.dev, self.address['network'])])
             self.fw.append(
-                ["filter", "", "-A INPUT -i %s -p tcp -m tcp --dport 53 -j ACCEPT" % self.dev])
+                ["filter", "", "-A INPUT -i %s -p tcp -m tcp --dport 53 -s %s -j ACCEPT" % (self.dev, self.address['network'])])
```

Comment:
@jburwell there were several places where such minor refactoring could be done, I've kept the scope limited to the change in the PR. @NuxRo I've refactored it as per John's comment. I'll see if we've an existing marvin test for this, otherwise I'll add one tomorrow/later-this-week.